backend/vehicle/urls.py

```diff
@@ -1,10 +1,16 @@
 from django.contrib import admin
 from django.urls import include, path
 
-from .views import VehicleDetailApiView, VehicleFilterList, VehicleListApiView
+from .views import (
+    VehicleDetailApiView,
+    VehicleFilterList,
+    VehicleListApiView,
+    SaveUnitApiView,
+)
 
 urlpatterns = [
     path("", VehicleListApiView.as_view(), name="vehicle"),
     path("filter/", VehicleFilterList.as_view(), name="vehicle-list"),
+    path("save/", SaveUnitApiView.as_view(), name="save-unit-to-list"),
```

Comment:
I'd suggest refactoring the urls into generic names, since adding an extra url param is redundant considering we can just use request.user, it would save a lot more time
Additionally, I think we should also add in some sort of context like "save/" so that we can clearly identify what the url endpoint is used for

Reply:
@kevinrczhang These are actually RESTful API design conventions that many companies use. We want to try to follow these conventions as much as possible.

Reply:
@kevinrczhang For the second point, a `POST` request is essentially a save/create request. So, the POST request already implicitly says that you are saving the given vehicle to a user.

Reply:
the other thing is because i merged the two tickets together for adding and removing from a list (reflected in my changes). I could separate them into different views however that would result in duplicated code, do we want to do that or keep it as is?

Reply:
@kevinrczhang Yeah, we really shouldn't be doing that. Using a flag to determine deleting/saving introduces a lot of coupling between the backend and frontend (i.e. the frontend will now need to know a lot about the implementation of the backend). I think the url should be the same, but saving shoud be `POST` and removing under `DELETE`.

Reply:
Alright, cool then.
i'll just add it under this PR anyways since the delete branch would rely on this branch